roles/create_mce_crs/tasks/main.yml

```diff
@@ -0,0 +1,193 @@
+  kubernetes.core.k8s:
+    state: present
+    kubeconfig: "{{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+    definition:
+      apiVersion: extensions.hive.openshift.io/v1beta1
+      kind: AgentClusterInstall
+      metadata:
+        name: "{{ namespace }}"
+        namespace: "{{ namespace }}"
+      spec:
+        apiVIP: "{{ api_vip }}"
+        clusterDeploymentRef:
+          name: "{{ namespace }}"
+        imageSetRef:
+          name: "openshift-v{{ openshift_full_version }}"
+        ingressVIP: "{{ ingress_vip }}"
+        platformType: "{{ platform_type }}"
+        networking:
+          clusterNetwork:
+          - cidr: "{{ cluster_network_cidr }}"
+            hostPrefix: 23
+          serviceNetwork:
+          - "{{ service_network_cidr }}"
+        provisionRequirements:
+          controlPlaneAgents: 3
+        sshPublicKey: "{{ sshkey }}"
+
+- name: Create Kubeconfig
+  command: "oc -n {{ namespace }} create secret generic {{ namespace }}-admin-kubeconfig --from-file=kubeconfig={{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+
+- name: Set password from day2_password
+  set_fact:
+    password:  "{{ day2_password }}"
+  when: day2_password is defined
+
+- name: Lookup day2 password from vault
+  when: day2_password is not defined
+  block:
+    - name: Decrypt vault
+      shell: ansible-vault ---decrypt {{ kubeadmin_vault_path }}
+      when: "{{  kubeadmin_vault_password_file_path is defined}}"
+
+    - name: Set Password from kubeadmin vault
+      set_fact:
+        password: "{{ (lookup(file, kubeadmin_vault_path) | from_yaml).password }}"
+
+  rescue:
+    - name: Get creds from assisted installer
+      uri:
+        url: "{{ URL_ASSISTED_INSTALLER_CLUSTER }}/credentials"
+        return_content: yes
+      register: credentials
+
+    - name: Set fact from assited installer creds
+      set_fact:
+        password: "{{  credentials.json.password }}"
+
+- name: Create kubeadmin user password
+  command: "oc -n {{ namespace }} create secret generic {{ namespace }}-admin-password --from-literal=username={{ day2_user }} --from-literal=password={{ password }}"
+
+- name: Create  Import Cluster (ClusterDeployment)
+  kubernetes.core.k8s:
+    state: present
+    kubeconfig: "{{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+    definition:
+      apiVersion: hive.openshift.io/v1
+      kind: ClusterDeployment
+      metadata:
+        name: "{{ namespace }}"
+        namespace: "{{ namespace }}"
+      spec:
+        baseDomain: "{{ base_dns_domain }}"
+        clusterInstallRef:
+          group: extensions.hive.openshift.io
+          kind: AgentClusterInstall
+          name: "{{ namespace }}"
+          version: v1beta1
+        clusterMetadata:
+          adminKubeconfigSecretRef:
+            name: "{{ namespace }}-admin-kubeconfig"
+          adminPasswordSecretRef:
+            name: "{{ namespace }}-admin-password"
+          clusterID: "{{ clusterID }}"
+          infraID: "{{ infraID }}"

```

Comment:
Where is value coming from?

Reply:
These 2 are coming from these commands, they need to be collected after cluster is deployed:

infraID: oc get infrastructure cluster -o json | jq .status.infrastructureName
clusterID: oc get clusterversion version -o json | jq .spec.clusterID

Reply:
So we could get these values using the k8s module? Are there tasks for this?

Reply:
We dont have tasks for this. These variables are set in inventory file.

Reply:
Added a task for this in that role